c_glib/gandiva-glib/function-registry.cpp

```diff
@@ -0,0 +1,179 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <gandiva-glib/function-registry.hpp>
+#include <gandiva-glib/native-function.hpp>
+
+G_BEGIN_DECLS
+
+/**
+ * SECTION: function_registry
+ * @short_description: FunctionRegistry class
+ */
+
+typedef struct GGandivaFunctionSignaturePrivate_ {
+  std::shared_ptr<gandiva::FunctionRegistry> *function_registry;
+  GList *native_function_list;
+} GGandivaFunctionRegistryPrivate;
```

Comment:
We don't need to this because we don't keep `gandiva::FunctionRegistry` instance and `native_function_list`.